apps/omg_child_chain/lib/omg_child_chain/fee_server.ex

```diff
@@ -95,37 +104,45 @@ defmodule OMG.ChildChain.FeeServer do
     new_state =
       case update_fee_specs(state) do
         {:ok, updated_state} ->
-          Alarm.clear(Alarm.invalid_fee_file(__MODULE__))
+          Alarm.clear(Alarm.invalid_fee_source(__MODULE__))
           updated_state
 
         :ok ->
-          Alarm.clear(Alarm.invalid_fee_file(__MODULE__))
+          Alarm.clear(Alarm.invalid_fee_source(__MODULE__))
           state
 
         _ ->
-          Alarm.set(Alarm.invalid_fee_file(__MODULE__))
+          Alarm.set(Alarm.invalid_fee_source(__MODULE__))
           state
       end
 
     {:noreply, new_state}
   end
 
-  @spec update_fee_specs(map()) :: :ok | {:ok, map()} | {:error, list({:error, atom(), any(), non_neg_integer() | nil})}
-  defp update_fee_specs(%{fee_adapter: adapter, expire_fee_timer: current_timer} = state) do
+  @spec update_fee_specs(t()) :: :ok | {:ok, t()} | {:error, list({:error, atom(), any(), non_neg_integer() | nil})}
+  defp update_fee_specs(
+         %__MODULE__{
+           fee_adapter: fee_adapter,
+           fee_adapter_opts: fee_adapter_opts,
+           expire_fee_timer: current_expire_fee_timer,
+           fee_buffer_duration_ms: fee_buffer_duration_ms
+         } = state
+       ) do
     source_updated_at = :ets.lookup_element(:fees_bucket, :fee_specs_source_updated_at, 2)
+    current_fee_specs = load_current_fees()
 
-    case adapter.get_fee_specs(source_updated_at) do
+    case fee_adapter.get_fee_specs(fee_adapter_opts, current_fee_specs, source_updated_at) do
       {:ok, fee_specs, source_updated_at} ->
         :ok = save_fees(fee_specs, source_updated_at)
-        _ = Logger.info("Reloaded fee specs from #{inspect(adapter)}, changed at #{inspect(source_updated_at)}")
-        new_timer = start_expiration_timer(current_timer)
+        _ = Logger.info("Reloaded fee specs from #{inspect(fee_adapter)}, changed at #{inspect(source_updated_at)}")
+        new_expire_fee_timer = start_expiration_timer(current_expire_fee_timer, fee_buffer_duration_ms)
 
         _ =
           Logger.info(
-            "Timer started: previous fees will still be valid for #{inspect(@fee_buffer_duration_ms)} ms, or until new fees are set"
+            "Timer started: previous fees will still be valid for #{inspect(fee_buffer_duration_ms)} ms, or until new fees are set"
           )
 
-        {:ok, Map.put(state, :expire_fee_timer, new_timer)}
+        {:ok, %__MODULE__{state | expire_fee_timer: new_expire_fee_timer}}
 
       :ok ->
```

Comment:
you;re calling `update_fee_specs /1` from init, init is expecting {:ok, state}, here, you're returning just :ok... whats going on here? is this expected?

Reply:
In the snippet above your comment you can see that {:ok, state} is returned.
https://github.com/omisego/elixir-omg/blob/70e2c69b8d73c334d91c3a9952e2be1b2b781b4d/apps/omg_child_chain/lib/omg_child_chain/fee_server.ex#L145

In fact the returned value is
* :ok - when fees did not change significantly from already stored. In init this case is impossible because we reset ets
* {:ok, state} - when fees were updated
* error - otherwise

The last 2 we have already discussed.